Map/Include/SquareMap.hpp

```diff
@@ -14,14 +14,8 @@ class SquareMap : public Map
     char printField(const std::unique_ptr<Field>& field);
     void updateVisilibity(const Position& playerPosition);
     std::string getMapToPrint(const Position& playerPosition);
-    bool isMovePossible(const Position&, const Direction);
```

Comment:
źle to zrozumiałeś. Te motody maja zostać w tej klasie. Klasa Map.hpp narzuca nam tylko interfejs. Mówi nam o tym, że powinniśmy w klasie dziedziczącej mieć metode isMovePossible, która zwraca boola.

Map.hpp
virtual bool isMovePossible(const Position&, const Direction) = 0;

SquareMap.hpp
bool isMovePossible(const Position&, const Direction) override;

SquareMap.cpp
bool SquareMap::isMovePossible(const Position&, const Direction) 
{
implementacja
}

a Map.cpp nie powinno istnieć.

Reply:
makeFieldEmpty też powinno być jako virtual w Map.hpp i zaimplementowane w SquareMap?

Reply:
tak, możesz tak to zrobić